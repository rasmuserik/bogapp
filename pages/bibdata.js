import React from 'react';
import {app} from '../components/app';
import Work from '../components/work';
import {connect} from 'react-redux';
import Immutable from 'immutable';

function mapStateToProp(state) {
  let id = state.getIn(['query', 'id'], undefined);
  let bib = state.getIn(['works', id]);

  if (Immutable.isImmutable(bib)) {
    bib = bib.toJS();
  }

  return {id, bib};
}

export default app(
  connect(mapStateToProp)(({bib}) => <Work bib={bib} />)
);
