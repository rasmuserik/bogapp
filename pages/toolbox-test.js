import React from 'react';
import {app} from '../components/app';
import Button from 'react-toolbox/lib/button/Button';

export function ToolboxTest (props) {
  return <div>
    <Button label="hello" raised primary />
    <Button label="hi" accent />
    <Button label="hello" raised />
  </div>
}

export default app(ToolboxTest);
