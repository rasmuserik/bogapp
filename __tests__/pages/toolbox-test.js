import React from 'react';
import renderer from 'react-test-renderer';

import {ToolboxTest} from '../../pages/toolbox-test';

it('renders correctly', () => {
  const tree = renderer.create(<ToolboxTest />).toJSON();
  expect(tree).toMatchSnapshot();
});
