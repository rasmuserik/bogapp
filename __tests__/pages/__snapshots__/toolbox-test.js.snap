// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    className=""
    data-react-toolbox="button"
    disabled={false}
    href={undefined}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    hello
  </button>
  <button
    className=""
    data-react-toolbox="button"
    disabled={false}
    href={undefined}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    hi
  </button>
  <button
    className=""
    data-react-toolbox="button"
    disabled={false}
    href={undefined}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    hello
  </button>
</div>
`;
